File: FullTextSearch/src/test/java/SetOperators/SubtractSetOperatorTest.java
package SetOperators;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

import static org.junit.jupiter.api.Assertions.*;

class SubtractSetOperatorTest {

    @Test
    public void notEmptyInitialSetCommonElementTest() {
        Set<String> result = new HashSet<>(Arrays.asList("a", "b", "c"));
        Set<String> subMask = new HashSet<>(Arrays.asList("a", "b"));
        new SubtractSetOperator().specificOperation(result, subMask);
        org.assertj.core.api.Assertions.assertThat(result).containsOnly("c");
    }

    @Test
    public void notEmptyInitialSetSubtractNothing() {
        Set<String> result = new HashSet<>(Arrays.asList("a", "b", "c"));
        Set<String> subMask = new HashSet<>(Arrays.asList());
        new SubtractSetOperator().specificOperation(result, subMask);
        org.assertj.core.api.Assertions.assertThat(result).containsOnly("a", "b", "c");
    }

    @Test
    public void notEmptyInitialSetSubtractNotCommon() {
        Set<String> result = new HashSet<>(Arrays.asList("a", "b", "c"));
        Set<String> subMask = new HashSet<>(Arrays.asList("d", "e"));
        new SubtractSetOperator().specificOperation(result, subMask);
        org.assertj.core.api.Assertions.assertThat(result).containsOnly("a", "b", "c");
    }

    @Test
    public void EmptyInitialSetTest() {
        Set<String> result = new HashSet<>();
        Set<String> subMask = new HashSet<>(Arrays.asList("z", "y"));
        new SubtractSetOperator().specificOperation(result, subMask);
        org.assertj.core.api.Assertions.assertThat(result).doesNotContainSequence("z", "y");
    }

}

File: FullTextSearchCsharp/FullTextSearch/FullTextSearch.Test/SetOperators/SubtractSetOperatorTest.cs
using System.Collections.Generic;
using FullTextSearch.SetOperators;
using NFluent;
using Xunit;


namespace FullTextSearch.Test.SetOperators
{
    public class SubtractSetOperatorTest : SetOperatorTest
    {
        [Fact]
        public void NotEmptyInitialSetCommonElementTest()
        {
            ISet<string> result = new HashSet<string> {"a", "b", "c"};
            ISet<string> subMask = new HashSet<string> {"a", "b"};

            new SubtractSetOperator().SpecificOperation(result, subMask);
            Check.That(result).IsEquivalentTo("c");
        }

        [Fact]
        public void NotEmptyInitialSetSubtractNothing()
        {
            ISet<string> result = new HashSet<string> {"a", "b", "c"};
            ISet<string> subMask = new HashSet<string>();

            new SubtractSetOperator().SpecificOperation(result, subMask);
            Check.That(result).IsEquivalentTo("a", "b", "c");
        }

        [Fact]
        public void NotEmptyInitialSetSubtractNotCommon()
        {
            ISet<string> result = new HashSet<string> {"a", "b", "c"};
            ISet<string> subMask = new HashSet<string> {"d", "e"};

            new SubtractSetOperator().SpecificOperation(result, subMask);
            Check.That(result).IsEquivalentTo("a", "b", "c");
        }

        [Fact]
        public void EmptyInitialSetTest()
        {
            ISet<string> result = new HashSet<string>();
            ISet<string> subMask = new HashSet<string> {"d", "e"};

            new SubtractSetOperator().SpecificOperation(result, subMask);
            Check.That(result).Not.IsEquivalentTo("d", "e");
        }

        [Fact]
        public void TestFullFunctionalityOfSubtractOnInvertedIndex()
        {
            var initial = IndexMap["hello"];
            var result =
                new SubtractSetOperator().Operate(initial, new List<string> {"hi", "world"}, Indexes);
            Check.That(result).IsEquivalentTo("1");
        }
    }
}